unison/cell.go

```diff
@@ -0,0 +1,114 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package unison
+
+// Cell stores some state of type interface{}.
+// Intermittent updates are lost, in case the Cell is updated faster than the
+// consumer tries to read for state updates. Updates are immediate, there will
+// be no backpressure applied to producers.
+//
+// In case the Cell is used to transmit updates without backpressure, the
+// absolute state must be computed by the producer beforehand.
+//
+// A typical use-case for cell is to generate asynchronous configuration updates (no deltas).
+type Cell struct {
+	// All writes/reads to any of the internal fields must be guarded by mu.
+	mu Mutex
```

Comment:
Why not use `RWMutex`?

Reply:
We always modify state. The `readID` is modified by `Get` and `Wait`. The `writeID` is modified `Set`. And finally `waiter` and `state` are read and modified by all these methods and therefore must always be protected.